app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/ui/summary/AddonSummaryDestination.kt

```diff
@@ -0,0 +1,287 @@
+    }
+    var showConfirmationDialog by remember { mutableStateOf(false) }
+    if (showConfirmationDialog) {
+      HedvigAlertDialog(
+        title = stringResource(R.string.ADDON_FLOW_CONFIRMATION_TITLE),
+        onDismissRequest = { showConfirmationDialog = false },
+        onConfirmClick = onConfirmClick,
+        buttonSize = DialogDefaults.ButtonSize.BIG,
+        confirmButtonLabel = stringResource(R.string.ADDON_FLOW_CONFIRMATION_BUTTON),
+        dismissButtonLabel = stringResource(R.string.general_close_button),
+        text = stringResource(
+          R.string.ADDON_FLOW_CONFIRMATION_DESCRIPTION,
+          formattedDate,
+        ),
+      )
+    }
+    SummaryCard(
+      uiState = uiState,
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp),
+    )
+    Spacer(Modifier.weight(1f))
+    Spacer(Modifier.height(16.dp))
+    Column(
+      Modifier
+        .padding(horizontal = 16.dp),
+    ) {
+      HedvigNotificationCard(
+        message = stringResource(R.string.ADDON_FLOW_SUMMARY_INFO_TEXT),
+        priority = NotificationDefaults.NotificationPriority.Info,
+      )
+      Spacer(Modifier.height(40.dp))
+      HorizontalItemsWithMaximumSpaceTaken(
+        startSlot = {
+          HedvigText(
+            stringResource(R.string.TIER_FLOW_TOTAL), // todo: change copy prob?
+            style = HedvigTheme.typography.bodySmall,
+          )
+        },
+        spaceBetween = 8.dp,
+        endSlot = {
+          HedvigText(
+            text = stringResource(
+              R.string.ADDON_FLOW_PRICE_LABEL,
+              uiState.summaryParameters.quote.price.amount.toInt(),
+            ),
+            textAlign = TextAlign.End,
+            style = HedvigTheme.typography.bodySmall,
+          )
+        },
+      )
+      Row(
+        Modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.End,
+      ) {
+        HedvigText(
+          text = stringResource(R.string.ADDON_FLOW_SUMMARY_PRICE_SUBTITLE),
+          textAlign = TextAlign.End,
+          style = HedvigTheme.typography.label,
+          color = HedvigTheme.colorScheme.textSecondary,
+        )
+      }
+      Spacer(Modifier.height(16.dp))
+      HedvigButton(
+        text = stringResource(R.string.ADDON_FLOW_SUMMARY_CONFIRM_BUTTON),
+        modifier = Modifier.fillMaxWidth(),
+        buttonStyle = Primary,
+        buttonSize = Large,
+        enabled = true,
+        onClick = {
+          showConfirmationDialog = true
+        },
+      )
+      Spacer(Modifier.height(16.dp))
+    }
+  }
+}
+
+@Composable
+private fun SummaryCard(uiState: Content, modifier: Modifier = Modifier) {
+  val locale = getLocale()
+  val formattedDate = remember(uiState.summaryParameters.activationDate, locale) {
+    HedvigDateTimeFormatterDefaults.dateMonthAndYear(
+      locale,
+    ).format(uiState.summaryParameters.activationDate.toJavaLocalDate())
+  }
+  QuoteCard(
```

Comment:
I tested here, and I got in a scenario where my hard had "Show details" but there was nothing in the details itself. I wonder if this is just something which should never happen, or if we should guard against this state and not show the button at all and not make this clickable if that state is passed in.
<img width="521" alt="image" src="https://github.com/user-attachments/assets/1b95d6f3-fbe1-4a32-a3f6-d81d23435f92">

Reply:
Thought about this as well, but I think we should always get at least one of the documents there - TnA. So probably not worth safe-guarding, it's just a temporary staging thing now with the docs.
But may be we should log error if the docs list is empty?

Reply:
Yeah a log could be good, but yeah let's keep it this way then